lib/features/registration/pages/credit_card_details_page.dart

```diff
@@ -15,54 +15,68 @@ class CreditCardDetailsPage extends StatelessWidget {
     return Scaffold(
       body: BlocBuilder<StripeCubit, StripeState>(
         builder: (_, state) {
-          final response = state.stripeObject;
-
+          if (state.stripeStatus == StripeObjectStatus.initial) {
+            context.read<StripeCubit>().fetchSetupIntent();
+            return const Center(child: CircularProgressIndicator());
+          }
           if (state.stripeStatus == StripeObjectStatus.loading) {
             return const Center(child: CircularProgressIndicator());
           }
           if (state.stripeStatus == StripeObjectStatus.failure) {
             return const Center(child: Text('Could not connect to Stripe'));
           }
-          return Padding(
-            padding: const EdgeInsets.only(top: 30),
-            child: InAppWebView(
-              initialUrlRequest: URLRequest(
-                url: Uri.parse(
-                  response.url,
-                ),
+
+          if (state.stripeStatus != StripeObjectStatus.display) {
+            return const Center(child: CircularProgressIndicator());
+          }
+
+          final stripe = context.read<StripeCubit>().state.stripeObject;
+
+          Stripe.instance
+              .initPaymentSheet(
+            paymentSheetParameters: SetupPaymentSheetParameters(
```

Comment:
same here - looks like similar code above, so perhaps make it a widget can help?